### PR TITLE
feat : Program to find Kth Symbol of Grammar

### PR DESCRIPTION
**Pull request by hecker1002**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [ ] This pull request is all my own work -- I have not plagiarized.
- [ ] I know that pull requests will not be merged if they fail the automated tests.
- [ ] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [ ] All new JavaScript files are placed inside an existing directory.
- [ ] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [ ] All new algorithms have a URL in their comments that points to Wikipedia or another similar explanation.
- [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: Recursive/TowerOfHanoi.js [hunk 1]
```diff
@@ -14,5 +14,5 @@ export function TowerOfHanoi (n, from, to, aux, output = []) {
 
 // Driver code (A, C, B are the name of rods)
 
-// const n = 4
-// TowerOfHanoi(n, 'A', 'C', 'B')
+const n = 4
```
Comment (appgurueu) [inline Recursive/TowerOfHanoi.js:17]: Unrelated change.

File: Recursive/KSymbolOfGrammar.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      return KSymbol(n-1,k);
+    }
+    
+    else {
+      return !KSymbol(n-1 , k-mid)
+    }
+  }
+  
+  // Enter the row no. and col no. of matrix 
+  let ans = KSymbol( 4, 1)
```
Comment (appgurueu) [inline Recursive/KSymbolOfGrammar.js:27]: This should be in a test I assume.

File: Recursive/KSymbolOfGrammar.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// Recursive call function  to find l=k th Symbol of Grammar  (0 => 0,1 , 1=> 1,0)
+//   Extracted Matrix
+//    0
+//    0 1
+//    0 1 1 0
+//    0 1 1 0 1 0 0 1 
+//    0 1 ,.....
+
+function KSymbol(n ,k){
```
Comment (appgurueu) [inline Recursive/KSymbolOfGrammar.js:9]: needs more comments, also doesn't abide by our style guidelines

File: Recursive/KSymbolOfGrammar.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// Recursive call function  to find l=k th Symbol of Grammar  (0 => 0,1 , 1=> 1,0)
```
Comment (appgurueu) [inline Recursive/KSymbolOfGrammar.js:1]: Please rephrase this

**Review by appgurueu — CHANGES_REQUESTED** [review]
Missing proper tests, CI is failing.

**Comment by Mohammed-Faizzzz** [comment]
Hi, may I check if help is still required? Please let me know, I would be very much willing to help!

**Comment by appgurueu** [comment]
> Hi, may I check if help is still required? Please let me know, I would be very much willing to help!

Well, the issues still haven't been resolved.

**Pull request closed without merging**